tests/integration_tests/tests/agent_tests/test_snapshots.py

```diff
@@ -103,23 +126,57 @@ def _restore_snapshot_multitenant(self, snapshot_id):
     def test_snapshot_with_agents(self):
         states = [AgentState.STARTED, AgentState.CREATING]
         deployments = self._deploy_with_agents(states)
-        self.assertEqual(len(self.client.agents.list().items), 1)
+        self.assertEqual(len(self.client.agents.list().items), 2)
         snapshot_id = self._create_snapshot()
-        self._undeploy(states, deployments)
+
+        downloaded_snapshot = os.path.join(self.workdir, 'snapshot.zip')
+        self.client.snapshots.download(
+            snapshot_id,
+            output_file=downloaded_snapshot,
+        )
+
+        reset_storage(self.env.container_id)
+        self.client.snapshots.upload(downloaded_snapshot, snapshot_id)
         self._restore_snapshot(states, deployments, snapshot_id)
-        self.assertEqual(len(self.client.agents.list().items), 1)
+
+        # unlike the other tenant tests, in this one we FIRST reset + restore,
+        # and only then undeploy. This checks that agent connectivity is
+        # still available after that reset + restore
+        self.assertEqual(len(self.client.agents.list().items), 2)
```

Comment:
It doesn't assert that with your changes though, because if they're not connected you'll still get two items?

Reply:
the undeploy being successful (not throwing an exception) checks that. Without the other changes, either way, this one was just checking the in-db state (just listing agents never checks connectivity)